romancal/tweakreg/tests/test_astrometric_utils.py

```diff
@@ -481,39 +598,51 @@ def test_get_catalog_valid_parameters_but_no_sources_returned():
 def test_get_catalog_using_epoch(ra, dec, epoch):
     """Test that get_catalog returns coordinates corrected by proper motion."""
 
-    result = get_catalog(ra, dec, epoch=epoch)
+    result_all = get_catalog(ra, dec, epoch=epoch)
+
+    # select sources with reliable astrometric solutions based on the
+    # parallax_over_error parameter as discussed in Fabricius et al. 2021
+    # (https://www.aanda.org/articles/aa/full_html/2021/05/aa39834-20/aa39834-20.html)
+    mask = result_all["parallax_over_error"] > 5
+
+    result = result_all[mask]
     returned_ra = np.array(result["ra"])
     returned_dec = np.array(result["dec"])
 
     # get GAIA data and update coords to requested epoch using pm measurements
     gaia_ref_epoch = 2016.0
-    gaia_ref_epoch_coords = get_catalog(ra, dec, epoch=gaia_ref_epoch)
+    gaia_ref_epoch_coords_all = get_catalog(ra, dec, epoch=gaia_ref_epoch)
 
-    expected_new_dec = (
-        np.array(
-            gaia_ref_epoch_coords["dec"] * 3600
-            + (epoch - gaia_ref_epoch) * gaia_ref_epoch_coords["pmdec"] / 1000
-        )
-        / 3600
+    gaia_ref_epoch_coords = gaia_ref_epoch_coords_all[mask]
+
+    # calculate proper motion corrections
+    get_proper_motion_correction(
+        epoch=epoch,
+        gaia_ref_epoch_coords=gaia_ref_epoch_coords,
+        gaia_ref_epoch=gaia_ref_epoch,
     )
-    average_dec = np.array(
-        [
-            np.mean([new, old])
-            for new, old in zip(expected_new_dec, gaia_ref_epoch_coords["dec"])
-        ]
+    # calculate parallax corrections
+    get_parallax_correction(epoch=epoch, gaia_ref_epoch_coords=gaia_ref_epoch_coords)
+
+    # calculate the expected coordinates value after corrections have been applied to
+    # Gaia's reference epoch coordinates
+    expected_ra = (
+        gaia_ref_epoch_coords["ra"]
+        + gaia_ref_epoch_coords["pm_delta_ra"]
+        + gaia_ref_epoch_coords["parallax_delta_ra"]
     )
-    pmra = gaia_ref_epoch_coords["pmra"] / np.cos(np.deg2rad(average_dec))
-    expected_new_ra = (
-        np.array(
-            gaia_ref_epoch_coords["ra"] * 3600
-            + (epoch - gaia_ref_epoch) * (pmra / 1000)
-        )
-        / 3600
+    expected_dec = (
+        gaia_ref_epoch_coords["dec"]
+        + gaia_ref_epoch_coords["pm_delta_dec"]
+        + gaia_ref_epoch_coords["parallax_delta_dec"]
     )
 
     assert len(result) > 0
-    assert np.isclose(returned_ra, expected_new_ra, atol=1e-10, rtol=1e-9).all()
-    assert np.isclose(returned_dec, expected_new_dec, atol=1e-10, rtol=1e-9).all()
+    # choosing atol=0 corresponds to abs(a - b) / abs(b) <= rtol,
+    # where a is the returned value from the VO API and b is the expected
+    # value from applying the calculated PM and parallax corrections.
+    assert np.isclose(returned_ra, expected_ra, atol=0, rtol=1e-5).all()
+    assert np.isclose(returned_dec, expected_dec, atol=0, rtol=1e-5).all()
```

Comment:
An rtol of 1e-5 worries me?  For an object with a coordinate of something like 100, this is 1e-3 deg = 3.6" = 36 pixels?  Even for something with a coordinate of ~1, this is 0.36 pixels while we want to be doing more like 0.01 pixel.  Maybe just an atol of 0.001 and no rtol?

Reply:
Changed `atol` to `1e-5` and `rtol=0`.